x-pack/plugin/deprecation/qa/rest/src/main/java/org/elasticsearch/xpack/deprecation/TestDeprecationHeaderRestAction.java

```diff
@@ -77,19 +75,18 @@ public TestDeprecationHeaderRestAction(Settings settings) {
         this.settings = settings;
     }
 
-    @Override
-    public List<DeprecatedRoute> deprecatedRoutes() {
-        return singletonList(new DeprecatedRoute(GET, "/_test_cluster/deprecated_settings", DEPRECATED_ENDPOINT));
-    }
-
     @Override
     public String getName() {
         return "test_deprecation_header_action";
     }
 
     @Override
     public List<Route> routes() {
-        return Collections.emptyList();
+        return List.of(
+            Route.builder(GET, "/_test_cluster/deprecated_settings")
+                .deprecated(DEPRECATED_ENDPOINT, RestApiCompatibleVersion.currentVersion())
```

Comment:
might want a comment here stating that .currentVersion() is for testing purposes only and actual version should be used in real code. I know i will read the tests for examples of usage and that comment could help to prevent incorrect usage.

Reply:
c178121